src/main/java/de/tum/in/www1/artemis/service/connectors/gitlab/GitLabService.java

```diff
@@ -276,20 +276,25 @@ public void createProjectForExercise(ProgrammingExercise programmingExercise) th
         final var group = new Group().withPath(exercisePath).withName(exerciseName).withVisibility(Visibility.PRIVATE);
         try {
             gitlab.getGroupApi().addGroup(group);
-
-            final var instructors = userService.getInstructors(programmingExercise.getCourseViaExerciseGroupOrCourseMember());
-            final var teachingAssistants = userService.getTutors(programmingExercise.getCourseViaExerciseGroupOrCourseMember());
-            for (final var instructor : instructors) {
-                final var userId = gitLabUserManagementService.getUserId(instructor.getLogin());
-                gitLabUserManagementService.addUserToGroups(userId, List.of(programmingExercise), MAINTAINER);
+        }
+        catch (GitLabApiException e) {
+            if (e.getMessage().contains("has already been taken")) {
+                // ignore this error, because it is not really a problem
+                log.warn("Failed to add group " + exerciseName + " due to error: " + e.getMessage());
             }
-            for (final var ta : teachingAssistants) {
-                final var userId = gitLabUserManagementService.getUserId(ta.getLogin());
-                gitLabUserManagementService.addUserToGroups(userId, List.of(programmingExercise), GUEST);
+            else {
+                throw new GitLabException("Unable to create new group for course " + exerciseName, e);
             }
         }
-        catch (GitLabApiException e) {
-            throw new GitLabException("Unable to create new group for course " + exerciseName, e);
+        final var instructors = userService.getInstructors(programmingExercise.getCourseViaExerciseGroupOrCourseMember());
+        final var teachingAssistants = userService.getTutors(programmingExercise.getCourseViaExerciseGroupOrCourseMember());
+        for (final var instructor : instructors) {
+            final var userId = gitLabUserManagementService.getUserId(instructor.getLogin());
+            gitLabUserManagementService.addUserToGroups(userId, List.of(programmingExercise), MAINTAINER);
```

Comment:
@krusche Is it on purpose that this operation (and also the corresponding one for TAs) are no longer within a try-catch?

Reply:
yes, the method `gitLabUserManagementService.addUserToGroups(...)` itself has a try catch block and does not throw if the user is already part of the group. So I don't think there is a reason to have another try catch here, in particular because the previous catch (which I moved up) only re-threw the exception